### ISSUE TITLE
Add dockerfile?

### ISSUE THREAD

**Issue by Ognian** [sgielen/picl-k3os-image-generator#16]
Have you thought of adding a dockerfile so that this can run on a Mac or even Windows?
Ognian

**Comment by sgielen** [comment]
I'm not sure if we can, because we use root privileges to perform loopback mounts. I think a privileged container could perhaps work. Would you like to give it a go?

**Comment by Ognian** [comment]
Ok I tried this out and ended with the following:
```
...

2020-05-13 15:39:13 (5.54 MB/s) - 'deps/rpi-firmware-nonfree-master.zip' saved [110930819]

== Making image and filesystems... ==
/dev/mapper/control: open failed: No such device
Failure to communicate with kernel device-mapper driver.
Check that device-mapper is available in the kernel.
Incompatible libdevmapper 1.02.167 (2019-11-30) and kernel driver (unknown version).
device mapper prerequisites not met
```
If you want I can make a pull request with the work I did until now...

**Comment by Ognian** [comment]
I've added some debug output:
![image](https://user-images.githubusercontent.com/1231174/81945856-eab23200-95fe-11ea-8bd0-c77d85266aac.png)
and the output
![image](https://user-images.githubusercontent.com/1231174/81945926-01f11f80-95ff-11ea-9f08-9bad6f860578.png)

looks like parts -a /dev/loopX is failing...
any ideas on this?

**Comment by Ognian** [comment]
Maybe [this](https://github.com/moby/moby/issues/27886) can help...
I'm running privileged have devices /dev:/dev and cap_add: CAP_MKNOD

**Comment by Ognian** [comment]
it should work now...
added pull request #18

**Comment by Ognian** [comment]
@sgielen have you been able to review the PR already?

**Comment by sgielen** [comment]
Docker support landed in abd9ca3ec5f6bd2d007d3685f72396d86f8f150a, thanks @Ognian!
